design/specs/column-extraction.md

```diff
@@ -8,87 +8,66 @@ editor: markdown
+### Selecting columns as a starting point
 
-2. **The user selects the columns (attributes) for the new entity**
+The user will select one or multiple columns to start with the column extraction process. All columns can be extracted, except for primary key columns. Once selected, the inspector panel will list 'Extract Columns to Table' as an action. Selecting this option will open the 'Extract Columns to Table' dialog.
 
-    The user chooses all the columns that will be attributes for the new object. For example, in the instance of the publisher, there could be columns for 'Publisher Name' and 'Publisher Id.' Including all the attributes is very important since columns cannot be moved after the fact.
+![image](/assets/design/specs/column-extraction/185139636-9a8a048c-9f87-4e26-aa2c-88f4e8d1648d.png)
 
-     <img width="1512" alt="image" src="https://user-images.githubusercontent.com/845767/177544077-9f074830-ce64-4ed9-80bc-53de53977f9c.png">
+## Extract Columns Dialog
 
+The dialog will list the columns that will be extracted and the table that will be created. The user can change the name of the new table. The dialog will also list the links that will be created between the new table and the original table. The user can change the name of the link column.
 
-3. **The user clicks on the 'Extract Columns' option**
+![image](/assets/design/specs/column-extraction/185901982-f991a6a3-c228-4c94-9d57-55b921e44511.png)
 
-    The user can choose to keep copies of the columns in the table if needed. By default the extracted columns are deleted.
+### Impact of the Extract Columns Operation
 
-4. **The user inspects the new foreign key column (link)**
+Since the split operation will affect the original table, the user should be made aware of the operation's impact. The dialog should communicate the following:
 
-    The new foreign key column will contain links to the new records that have been created. Additionally, when selected, the inspector panel will display the link properties.
+- The original table will be updated to remove the extracted columns
+- A new table will be created with the extracted columns
+- A link will be created between the new table and the original table
 
-    <img width="1469" alt="image" src="https://user-images.githubusercontent.com/845767/177546415-23b11ded-8987-4e69-ab1a-1decb9e773d3.png">
+### Link Column Naming Convention
 
+It was decided that the link column naming convention should be as follows:
 
-5. **The user navigates to the new table**
+>In the application, where singularization is not straightforward, we don’t suggest column names. Instead we require the user to manually name the FK column.
```

Comment:
The below section `Using Rules to Generate Column Names` talks about using rules for generating column names to be a future feature. But here it is mentioned 
> where singularization is not straightforward, we don’t suggest column names

So Should we suggest the link column name if singularization is possible now?

Reply:
@ghislaineguerin This is not what we decided in the email thread, this was @seancolsen's original proposal. We decided that the link column's name would be based on the table name. So if the user enters `Author` as a table name, then the link column would also be called `Author` (unless the name is taken already, then we can leave it blank).

Also it would be helpful to link to the email thread (on the mailing list) from the spec so people can follow the reference back to the source.

Reply:
@kgodey I think instead of blank, a number suffix can be added to the link column name if the name is already taken. The number will be incremented by 1 for each new link column.